service-base/src/test/java/org/eclipse/hono/service/registration/AbstractRegistrationServiceTest.java

```diff
@@ -724,22 +650,17 @@ public void testDeleteSucceedWithMissingResourceVersion(final VertxTestContext c
     @Test
     public void testDeleteSucceedWithCorrectResourceVersion(final VertxTestContext ctx) {
 
-        final Promise<OperationResult<Id>> result = Promise.promise();
         final Checkpoint register = ctx.checkpoint(2);
 
-        getDeviceManagementService().createDevice(TENANT, Optional.of(DEVICE), new Device(), NoopSpan.INSTANCE, result);
-        result.future()
-        .map(response -> {
-            assertEquals(HttpURLConnection.HTTP_CREATED, response.getStatus());
-            register.flag();
-            return response;
-        }).compose(rr -> {
-            final Promise<Result<Void>> update = Promise.promise();
+        getDeviceManagementService().createDevice(TENANT, Optional.of(DEVICE), new Device(), NoopSpan.INSTANCE)
+                .map(response -> {
+                    assertEquals(HttpURLConnection.HTTP_CREATED, response.getStatus());
+                    register.flag();
+                    return response;
+                }).compose(rr -> {
             final String resourceVersion = rr.getResourceVersion().orElse(null);
-
-            getDeviceManagementService().deleteDevice(
-                    TENANT, DEVICE, Optional.of(resourceVersion), NoopSpan.INSTANCE, update);
-            return update.future();
+            return getDeviceManagementService()
+                    .deleteDevice(TENANT, DEVICE, Optional.of(resourceVersion), NoopSpan.INSTANCE);
```

Comment:
Indentation in this method seems non-standard.

Something which was there in the code before already: Since `resourceVersion` can be null, `Optional.ofNullable(resourceVersion)` should be used, I guess.